models/pattern/core/register.go

```diff
@@ -233,3 +236,75 @@ func registerMesheryServerOAM(rootPath string, constructs []string, regFn func([
 
 	return fmt.Errorf("%s", strings.Join(errs, "\n"))
 }
+
+func GetK8Components(config []byte, ctx string) (*manifests.Component, error) {
+	cli, err := kubernetes.New(config)
+	if err != nil {
+		return nil, err
+	}
+	req := cli.KubeClient.RESTClient().Get().RequestURI("/openapi/v2")
+	res := req.Do(context.Background())
+	content, err := res.Raw()
+	if err != nil {
+		return nil, err
+	}
+	apiResources, err := getAPIRes(cli)
+	if err != nil {
+		return nil, err
+	}
+	manifest := string(content)
+	man, err := manifests.GenerateComponents(manifest, manifests.K8s, manifests.Config{
+		Name: "Kubernetes",
+		Filter: manifests.CrdFilter{
+			IsJson:        true,
+			OnlyRes:       apiResources, //When crd or api-resource names are directly given, we dont need NameFilter
+			RootFilter:    []string{"$.definitions"},
+			VersionFilter: []string{"$[0]"},
+			GroupFilter:   []string{"$[0]"},
+			ItrFilter:     "$..[\"x-kubernetes-group-version-kind\"][?(@.kind",
+			ItrSpecFilter: "$[0][?(@[\"x-kubernetes-group-version-kind\"][0][\"kind\"]",
+			GField:        "group",
+			VField:        "version",
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return man, nil
+}
+
+func DeleteK8sWorkloads(ctx string) {
+	//Iterate through entire store
```

Comment:
Is this an in-memory store only? Or is this store the sqlite database?

Reply:
in memory store